rustainers/src/runner/inner.rs

```diff
@@ -287,6 +331,82 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
         Ok(())
     }
 
+    fn get_docker_host(&self) -> Option<String> {
+        env::var("DOCKER_HOST").ok()
+    }
+
+    #[tracing::instrument(skip(self),fields(runner = %self))]
+    async fn find_host_network(&self) -> Result<Option<Network>, ContainerError> {
+        // If we're docker in docker running on a custom network, we need to inherit the
+        // network settings, so we can access the resulting container.
+        let docker_host = self.host().await?;
+        let custom_networks = self.list_custom_networks().await?;
+        let mut networks = vec![];
+        for network in custom_networks {
+            let network_configs = self.list_network_config(network.id).await?;
+            networks.extend(
+                network_configs
+                    .into_iter()
+                    .filter_map(|x| {
+                        x.subnet.map(|x| {
+                            x.contains(IpAddr::V4(docker_host.0))
+                                .then(|| Network::Custom(network.name.clone()))
+                                .ok_or(0)
+                        })
+                    })
+                    // Chained filtermaps is not ideal
+                    .filter_map(std::result::Result::ok)
```

Comment:
🎯 **suggestion**: you can  add the `.ok()` at the end of line 354 to avoid the chained filtermap.

But IMHO, this code is too hard to read.
The method seems to return the first network that match the condition.
I think that a nested for-loop with an inner `return` should be easier to read.